### PR TITLE
Fix GradientDescent Opt to not require FiniteDiff type

### PR DESCRIPTION
**Pull request by dylanjm**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

#2175 

##### What are the significant changes in functionality due to this change request?

This PR removes the requirement for GradientDescent optimizer to be of `type="FiniteDifference"`

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by dylanjm** [comment]
@wangcj05 It looks like SimulatedAnnealing was removed in the patch. I'm not familiar with SA, is this just a type of GradientDescent? Or should I add that type back in?

File: ravenframework/Models/Model.py [hunk 1]
```diff
@@ -131,10 +131,7 @@ class cls.
   validateDict['Optimizer'][0]['class'       ] ='Optimizers'
   validateDict['Optimizer'][0]['required'    ] = False
   validateDict['Optimizer'][0]['multiplicity'] = 1
-  validateDict['Optimizer'][0]['type'] = ['SPSA',
-                                          'FiniteDifference',
-                                          'ConjugateGradient',
-                                          'SimulatedAnnealing',
+  validateDict['Optimizer'][0]['type'] = ['GradientDescent',
                                           'GeneticAlgorithm',
```
[the remark this answers is not in the document]
Comment (alfoa) [reply ravenframework/Models/Model.py:135]: sorry we missed this

**Comment by alfoa** [comment]
> @wangcj05 It looks like SimulatedAnnealing was removed in the patch. I'm not familiar with SA, is this just a type of GradientDescent? Or should I add that type back in?

sorry Daylan. I added a patch that you can apply. SimulatedAnnealing is supposed to be In the validation dictionary as well.

**Comment by alfoa** [comment]
@dylanjm also doc/workshop/optimizer/Inputs/1_grad_desc.xml  should be updated. Sorry I missed this.

**Review by PaulTalbot-INL — APPROVED** [review]
Code changes have been reviewed and are in accordance to standards.

**Comment by PaulTalbot-INL** [comment]
Only waiting on tests to pass before merging. If tests pass without further changes, feel free to merge.

**Review by alfoa — APPROVED** [review]

**Pull request merged**